.github/workflows/nestbuildmatrix.yml

```diff
@@ -2,32 +2,444 @@
 # detailed syntax defined in
 # https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions
 #
-name: "NEST matrix jobs"
-env:
-  CXX_FLAGS: "-pedantic -Wextra -Woverloaded-virtual -Wno-unknown-pragmas"
-  PYTHONPATH: ${{ github.workspace }}/build/python
-on: [push, pull_request]
+name: "Test suite"
```

Comment:
What is this name about? If it the title overarching *all* CI jobs, I think it should rather be something like "NEST CI jobs", as "Test suite" implies that this is running (only) our [testsuite](https://github.com/nest/nest-simulator/tree/master/testsuite).